### ISSUE TITLE
error when provisioning ruby when doing vagrant up

### ISSUE THREAD

**Issue by yina** [Furaha/ugandasms#39]
Following is the error message after running vagrant up.

==> default: Get:53 http://security.ubuntu.com trusty-security/main Translation-en [251 kB]
==> default: Ign https://oss-binaries.phusionpassenger.com trusty/main Translation-en_US
==> default: Hit http://us.archive.ubuntu.com trusty/restricted Translation-en
==> default: Ign https://oss-binaries.phusionpassenger.com trusty/main Translation-en
==> default: Hit http://us.archive.ubuntu.com trusty/universe Translation-en
==> default: Get:54 http://security.ubuntu.com trusty-security/multiverse Translation-en [2,570 B]
==> default: Get:55 http://security.ubuntu.com trusty-security/restricted Translation-en [3,206 B]
==> default: Get:56 http://security.ubuntu.com trusty-security/universe Translation-en [74.7 kB]
==> default: Ign http://us.archive.ubuntu.com trusty/main Translation-en_US
==> default: Ign http://us.archive.ubuntu.com trusty/multiverse Translation-en_US
==> default: Ign http://us.archive.ubuntu.com trusty/restricted Translation-en_US
==> default: Ign http://us.archive.ubuntu.com trusty/universe Translation-en_US
==> default: Fetched 5,443 kB in 5s (1,008 kB/s)
==> default: Reading package lists...
==> default:  
==> default: **\*  apt() install ruby2.3 ruby2.3-dev nodejs nginx-extras passenger
==> default:  
==> default: Reading package lists...
==> default: Building dependency tree...
==> default: Reading state information...
==> default: Package ruby2.3 is not available, but is referred to by another package.
==> default: This may mean that the package is missing, has been obsoleted, or
==> default: is only available from another source
==> default: E
==> default: : 
==> default: Package 'ruby2.3' has no installation candidate
==> default: E
==> default: : 
==> default: Unable to locate package ruby2.3-dev
==> default: E
==> default: : 
==> default: Couldn't find any package by regex 'ruby2.3-dev'
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

**Comment by yina** [comment]
actually errors start here.

==> default: E
==> default: : 
==> default: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/p/pcre3/libpcre3_8.31-2ubuntu2.1_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
==> default: E
==> default: : 
==> default: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/t/tiff/libtiff5_4.0.3-7ubuntu0.3_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
==> default: E
==> default: : 
==> default: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/graphite2/libgraphite2-3_1.2.4-1ubuntu1.1_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
==> default: E
==> default: : 
==> default: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/p/pcre3/libpcrecpp0_8.31-2ubuntu2.1_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
==> default: E
==> default: : 
==> default: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/t/tiff/libtiffxx5_4.0.3-7ubuntu0.3_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
==> default: E
==> default: : Failed to fetch http://us.archive.ubuntu.com/ubuntu/pool/main/a/apt/apt-transport-https_1.0.1ubuntu2.11_amd64.deb  404  Not Found [IP: 91.189.91.23 80]
==> default: 
==> default: E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/git/git-man_1.9.1-1ubuntu0.2_all.deb  404  Not Found [IP: 91.189.88.161 80]
==> default: 
==> default: E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/git/git_1.9.1-1ubuntu0.2_amd64.deb  404  Not Found [IP: 91.189.88.161 80]
==> default: 
==> default: E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/p/pcre3/libpcre3-dev_8.31-2ubuntu2.1_amd64.deb  404  Not Found [IP: 91.189.88.161 80]
==> default: 
==> default: E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/t/tiff/libtiff5-dev_4.0.3-7ubuntu0.3_amd64.deb  404  Not Found [IP: 91.189.88.161 80]

**Comment by arafatm** [comment]
that looks like a network issue. can you try pinging that url

**Comment by yina** [comment]
ping's are working and I disabled all my firewall software. any thoughts?

**Comment by arafatm** [comment]
try sudo apt-get update && sudo apt-get dist-upgrade

**Comment by arafatm** [comment]
If that works you should be able to `sudo bootstrap/default.sh`

**Comment by yina** [comment]
dumb questions... would you take a look at the logs. I think the user being created isn't right so privileges become messed up. thanks!

[vagrant.txt](https://github.com/Furaha/ugandasms/files/234456/vagrant.txt)

**Comment by arafatm** [comment]
Yeah, you're getting weird networking issues. Are you running this in vagrant? If so, you should be able to do a `vagrant provision` or `vagrant reload --provision` to rerun the scripts. 

Couple other things to try:
- make sure you git pull the latest copy off master just in case
- set the vagrant in bridged mode. Networking seems to work better than in NAT

**Comment by arafatm** [comment]
I don't think it's a user issue? If you're running this in vagrant then it defaults to root when running the provisioning scripts.

**Comment by yina** [comment]
I did forget to add that I had to keep bootstrap in the original root directory rather than copy it the vagrant directory as in the first line of the deploy script. Do you think this is the issue?

**Comment by arafatm** [comment]
Probably. I have to provide a dir structure to copy the template files. But that doesn't fix your 404 errors. You have a networking issue you have to get past first. Take a look at the script. Not doing anything crazy.

**Comment by yina** [comment]
after from sleuthing it has to do with this line in bootstrap.sh

```
if [ "$[$(date +%s) - $(stat -c %Z /var/cache/apt/pkgcache.bin)]" -ge 3600 ] || /usr/bin/find /etc/apt/* -cnewer /var/cache/apt/pkgcache.bin | /bin/grep . > /dev/null; then
```

**Comment by arafatm** [comment]
that line checks the datestamp when the last `apt-get update` was run. If
it's more than 60 mins then it does an update. Otherwise it assumes the
packages are relatively new.

Are you available after 5pm PST to do a quick video call so I can see what
you're seeing?

**Comment by yina** [comment]
that's what i figured. just went ahead and commented it out, let it run, and sort it out later.
